include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -24,6 +24,15 @@
 namespace seqan3::detail
 {
 
+//!\brief An enumerator for the different error types used during the backtracking.
+enum class error_type : uint8_t
+{
+    DELETION,
+    INSERTION,
+    MATCHMM,
+    NONE
```

Comment:
Also please document the options via inline comments e.g. `none \\!< this is a nice option`.